src/Hazelcast.Net/Clustering/Invocation.cs

```diff
@@ -233,14 +231,20 @@ public async ValueTask<bool> CanRetryAsync(Func<long> correlationIdProvider)
         {
             if (correlationIdProvider == null) throw new ArgumentNullException(nameof(correlationIdProvider));
 
-            if (_cancellationToken.IsCancellationRequested) return false;
+            // fast fail on cancel
+            if (_cancellationToken.IsCancellationRequested) 
+                return false;
+
+            // fast fail on timeout
+            if (Clock.Milliseconds - StartTime > _messagingOptions.RetryTimeoutSeconds * 1000)
+                return false;
```

Comment:
Minor: On timeout, Java client completes the invocation with the special `OperationTimeoutException`, I guess here, we are completing it with whatever exception that caused this check

Reply:
true - fixing